ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/KsqlAuthorizationProviderHandler.java

```diff
@@ -33,19 +35,22 @@ public class KsqlAuthorizationProviderHandler implements Handler<RoutingContext>
 
   private final WorkerExecutor workerExecutor;
   private final KsqlAuthorizationProvider ksqlAuthorizationProvider;
+  private final Pattern unauthenticatedpaths;
 
-  public KsqlAuthorizationProviderHandler(final WorkerExecutor workerExecutor,
+  public KsqlAuthorizationProviderHandler(final Server server,
       final KsqlAuthorizationProvider ksqlAuthorizationProvider) {
-    this.workerExecutor = workerExecutor;
+    this.workerExecutor = server.getWorkerExecutor();
     this.ksqlAuthorizationProvider = ksqlAuthorizationProvider;
+    this.unauthenticatedpaths = AuthenticationPluginHandler.getAuthorizationSkipPaths(
+        server.getConfig().getList(KsqlRestConfig.AUTHENTICATION_SKIP_PATHS_CONFIG));
   }
 
   @Override
   public void handle(final RoutingContext routingContext) {
 
     final String path = routingContext.normalisedPath();
 
-    if (KSQL_AUTHENTICATION_SKIP_PATHS.contains(path)) {
+    if (unauthenticatedpaths.matcher(path).matches()) {
```

Comment:
Oh wow, so AUTHENTICATION_SKIP_PATHS_CONFIG was not taken into account at all before.